docs/google/best_practices.md

```diff
@@ -58,7 +74,69 @@ using a error rate of 0.5% per gate, a circuit of depth 20 and width 20 could be
 at 0.995^(20 * 20) = 0.135. Using separate error rates per gates (i.e. based on calibration
 metrics) or a more complicated noise model can result in more accurate error estimation.
 
-## Use sweeps when possible
+### Terminal Measurements
+
+Make sure that measurements are kept in the same moment as the final moment in
+the circuit.  Make sure that any circuit optimizers do not alter this by
+incorrectly pushing measurements forward. This behavior can be avoided by
+measuring all qubits with a single gate or by adding
+the measurement gate after all optimizers have run.
+
+Currently, only terminal measurements are supported by the hardware.  If you
+absolutely need intermediate measurements for your application, reach out to
+your Google sponsor to see if they can help devise a proper circuit using
+intermediate measurements.
```

Comment:
there's no checks against this and in fact you can get intermediate measurements to work (where work is defined as return bits without crashing)

Reply:
Yeah, but I don't think we want people doing this unless they have help.  It will be difficult to get working correctly (where correctly is defined as getting return bits that are not random noise).